src/v/cluster/types.cc

```diff
@@ -498,6 +498,19 @@ std::istream& operator>>(std::istream& i, replication_factor& cs) {
     return i;
 };
 
+replication_factor
+parsing_replication_factor(const std::optional<ss::sstring>& value) {
+    auto raw_value = boost::lexical_cast<int32_t>(*value);
```

Comment:
if this function has a pre-condition that the optional contains a value, then don't pass in the optional--pass in a const-ref to the string.